reader.go

```diff
@@ -80,7 +80,8 @@ func (r *Reader) ReadAt(p []byte, off int64) (int, error) {
 	}
 
 	if off+int64(l) > r.Length() {
-		return 0, errors.New("read beyond end of file")
+		l = int(r.Length() - off)
+		p = p[:l]
```

Comment:
I think it'll still act unusual here, and attempt to slice `p` by `[:-1]` before we get to 123 and panic.